### ISSUE TITLE
Semantic footer links

### ISSUE THREAD

**Issue by chris-gds** [Financial-Times/origami#1772]
## What

👋🏼 The footer links ideally should be `ol > li` elements due to the visual design and structure

## Why

Impacting WCAG 1.3.1

https://github.com/Financial-Times/origami/blob/7547d58077ac0fc42efc7a9dc88c671073ab9d5c/components/o-footer/demos/src/footer.mustache#L13

https://github.com/Financial-Times/origami/blob/7547d58077ac0fc42efc7a9dc88c671073ab9d5c/components/o-footer/demos/src/footer.mustache#L17

## Screenshots

<img width="1195" alt="FT footer links" src="https://github.com/user-attachments/assets/5ec3486b-4bd1-46a8-ba20-5e0472caa982">

## Anything else

[Improve semantics](https://www.w3.org/WAI/tutorials/menus/structure/#unordered-list)
